### DISCUSSION TITLE
battery.py conditions to left float mode

### DISCUSSION THREAD

**Discussion by ogurevich** [Louisvdw/dbus-serialbattery#650] [Q&A] [answered]
In the manage_charge_voltage_linear function (rows 217-222), there is a condition that sets self.allow_max_voltage = True if the following conditions are met:

voltageDiff >= utils.CELL_VOLTAGE_DIFF_TO_RESET_VOLTAGE_LIMIT
and not self.allow_max_voltage

my question is, what if the battery cells are still balanced and voltageDiff is still less than CELL_VOLTAGE_DIFF_TO_RESET_VOLTAGE_LIMIT? It seems that the battery will never leave the float mode. You're wondering if there is a misconception in your understanding.

If the battery cells are still balanced and the voltageDiff is below the CELL_VOLTAGE_DIFF_TO_RESET_VOLTAGE_LIMIT, the self.allow_max_voltage flag will remain False, and the battery will continue to operate in float mode.

**Comment by mr-manuel** [comment, marked as answer]
To which version are you referring?

Did you check this in the latest dev version?

https://github.com/Louisvdw/dbus-serialbattery/blob/2f6ea71ef606c9aabdc253abb8ef16df3dd92f74/etc/dbus-serialbattery/battery.py#L221-L225

See also https://github.com/Louisvdw/dbus-serialbattery/issues/635

**Comment by ogurevich** [comment]
thank you a lot, sorry, forgot to write the version: v1.0.20230508beta or on master branch. 
this project is absolutely great job you do ! thank you !
